sidecar/config/flags.go

```diff
@@ -25,6 +25,7 @@ const (
 	configFlag              = "config"
 	registerFlag            = "register"
 	proxyFlag               = "proxy"
+	proxyAdapterFlag        = "proxy_adapter"
```

Comment:
Renaming `--discovery_backend` and `--rules_backend` is _not_ a breaking change because these flags were never released. Both are post-[v0.4.2](https://github.com/amalgam8/amalgam8/blob/v0.4.2/sidecar/config/flags.go).
Either way, let's make sure to resolve #446 before the next release, and with a different PR.